tests/test_contracts/test_contracts.py

```diff
@@ -746,3 +746,57 @@ def test_precondition_violation_in_representation_invariant() -> None:
         Student("Bob", 1001001000, -19)
 
     assert 'my_condition2 precondition "x > 0" was violated' in str(exception_info.value)
+
+
+def test_parameter_int_float_without_strict(disable_strict_numeric_types) -> None:
```

Comment:
These tests are good, but are only checking parameter values. You also modified the code to apply this constant to return type checking, and there should be separate tests for that too.